File: src/modules/todos/application/todos/get-todos/get-todos.query-handler.ts
import { QueryHandler } from '../../../../../common/application/messaging/query-handler';
import { Todo } from '../../../domain/todo';
import { TodoRepository } from '../../abstractions/todo-repository';
import { GetTodosQuery } from './get-todos.query';

export class GetTodosQueryHandler implements QueryHandler<GetTodosQuery, Todo[]> {
  constructor(private readonly repository: TodoRepository) {}

  handle(_query: GetTodosQuery): Promise<Todo[]> {
    return this.repository.getAll();
  }
}

File: api/src/Modules/Todos/Application/Todos/GetTodos/GetTodosQueryHandler.cs
namespace Checknote.Modules.Todos.Application.Todos.GetTodos;

using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using Checknote.Common.Application.Messaging;
using Checknote.Common.Domain;
using Checknote.Modules.Todos.Application.Abstractions;
using Checknote.Modules.Todos.Domain.Todos;

public sealed class GetTodosQueryHandler : IQueryHandler<GetTodosQuery, IReadOnlyCollection<Todo>>
{
    private readonly ITodoRepository todoRepository;

    public GetTodosQueryHandler(ITodoRepository todoRepository)
    {
        this.todoRepository = todoRepository;
    }

    public Task<Result<IReadOnlyCollection<Todo>>> Handle(
        GetTodosQuery query,
        CancellationToken cancellationToken)
    {
        return Task.FromResult(Result.Success(todoRepository.GetTodos()));
    }
}
